### PR TITLE
Move proxy to prow config generation

### PR DESCRIPTION
**Pull request by howardjohn**
This PR moves the proxy to the new prow config generation. I made this
enabled only for the "prow-stage" branch so we can do send a dummy PR to
validate I didn't break anything.

Benefits:
* Proxy follows same conventions of other repos (in terms of naming,
testgrid, branching, etc)
* Config should be much simpler to understand
* Resolves some issues with duplicate names, confusing context, etc

The only effective changes after this is the change to the job names. I
am open to renaming, the reason I didn't keep the job names as before is
we would have ended up with names like
proxy-postsubmit_proxy_postsubmit, due to the naming convention of
test_repo_jobtype.

### CODE REVIEW

**Comment by howardjohn** [comment]
I made this a draft to let the proxy maintainers take a look first

File: prow/config/jobs/proxy.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+support_release_branching: true
+image: gcr.io/istio-testing/prowbazel:0.5.16
+node_selector:
+  testing: build-pool
+branches: [prow-stage]
+
+jobs:
+- name: test
+  type: presubmit
+  repos: [istio/istio@master]
```
Comment (bianpengyuan) [inline prow/config/jobs/proxy.yaml]: Should these be istio/proxy?

**Comment by bianpengyuan** [comment]
This looks great. Thanks!

File: prow/config/jobs/proxy.yaml (hunk 1 above)
Comment (howardjohn) [reply prow/config/jobs/proxy.yaml]: `repos` are extra repos to checkout, proxy is implicitly checked out at the PR commit for presubmit and the commited PR for for postsubmit.

The proxy jobs all explicitly configured this before, so I assumed they depend on something in istio/istio, is that not needed?

Comment (bianpengyuan) [reply prow/config/jobs/proxy.yaml]: I don't think it is necessary. Maybe @PiotrSikora would have better idea. I think it is fine to remove it.

Comment (howardjohn) [reply prow/config/jobs/proxy.yaml]: Well this is just for a dummy test branch anyways, so I'll just remove it then add it back later if we need it. No risk of breaking something

Comment (bianpengyuan) [reply prow/config/jobs/proxy.yaml]: sg.

**Review by geeknoid — APPROVED** [review]

**Pull request merged**